src/BUILD

```diff
@@ -182,6 +183,23 @@ py_binary(
     "_with_jdk",
 ]]
 
+assert_size(
+    name = "assert_embedded_tools_size",
+    src = ":embedded_tools_srcs",
+    # WARNING: Only adjust the number in `expect` if you are intentionally
+    # adding or removing embedded tools. Know that the more embedded tools there
+    # are in Bazel, the bigger the binary becomes and the slower Bazel starts.
+    expect = 503,
+    margin = 5,  # percentage
+    visibility = ["//visibility:private"],
+)
+
+sh_test(
```

Comment:
can assert_size itself be a test? then you don't need this extra rule, and dummy.sh can be hidden away.

Reply:
This is an interesting approach. What was the reasoning behind? assert_size is a nice Skylark rule, it only feels weird to be used like this in a BUILD file. Can you create and run an assert_size() target from the sh test?

Reply:
> can assert_size itself be a test?

Yes it can! Done.

> Can you create and run an assert_size() target from the sh test?

That'd beat the purpose of analysis-time assertion and of not building the inputs. See the new comments in the docstring.